R/MsBackendMetaboLights.R

```diff
@@ -0,0 +1,267 @@
+#' skipped with `offline = TRUE` in which case only locally cached content
+#' is considered.
+#'
+#' @param object an instance of `MsBackendMetaboLights`.
+#'
+#' @param mtblsId `character(1)` with the ID of the MetaboLights data
+#'     set/experiment.
+#'
+#' @param assayId `character` with the file names of assay files of the data
+#'     set. If not provided (`assayId = character()`, the default), MS data
+#'     files of all data set's assays is loaded. Use
+#'     `mtbls_list_files(<MetaboLights ID>, pattern = "^a_")` to list all
+#'     available assay files of a data set `<MetaboLights ID>`.
+#'
+#' @param filePattern `character` with the pattern defining the supported (or
+#'     requested) file types. Defaults to
+#'     `filePattern = "mzML$|CDF$|cdf$|mzXML$"` hence restricting to mzML,
+#'     CDF and mzXML files supported by *Spectra*'s `MsBackendMzR` backend.
+#'
+#' @param offline `logical(1)` whether only locally cached content should be
+#'     evaluated/loaded.
+#'
+#' @param ... additional parameters; currently ignored.
+#'
+#' @details
+#'
+#' Data files are by default extracted from the column `"Derived Spectral
+#' Data File"` of the MetaboLights data set's *assay* table. If this column
+#' does not contain any supported file names, the assay's column
+#' `"Raw Spectral Data File"` is evaluated.
+#'
+#' The backend uses the
+#' [BiocFileCache](https://bioconductor.org/packages/BiocFileCache) package for
+#' caching of the data files. These are stored in the default local
+#' *BiocFileCache* cache along with additional metadata that includes the
+#' MetaboLights ID, the assay file name with which the data file is associated
+#' with. Note that at present only MS data files in *mzML*, *CDF* and *mzXML*
+#' format are supported.
+#'
+#' The `MsBackendMetaboLights` backend defines and provides additional spectra
+#' variables `"mtbls_id"`, `"mtbls_assay_name"` and
+#' `"derived_spectral_data_file"` that list the MetaboLights ID, the name of
+#' the assay file and the original data file name on the MetaboLights ftp
+#' server for each individual spectrum. The `"derived_spectral_data_file"` can
+#' be used for the mapping between the experiment/data sets samples and the
+#' individual data files, respective their spectra. This mapping is provided
+#' in the respective MetaboLights assay file.
+#'
+#' The `MsBackendMetaboLights()` is considered *read-only* and does thus not
+#' support changing *m/z* and intensity values directly.
+#'
+#' Also, merging of MS data of `MsBackendMetaboLights` is not supported and
+#' thus `c()` of several `Spectra` with MS data represented by
+#' `MsBackendMetaboLights` will throw an error.
+#'
+#' @importClassesFrom Spectra MsBackendMzR
+#'
+#' @exportClass MsBackendMetaboLights
+#'
+#' @author Philippine Louail, Johannes Rainer
+#'
+NULL
+
+setClass("MsBackendMetaboLights",
+         contains = "MsBackendMzR",
+         slots = c(mtblsId = "character",
+                   assays = "character"),
```

Comment:
valid point. I on purpose did not include a documentation of the `MsBackendMetaboLights` **class** because I don't want users to access the internal information in the class. Both the `@mtblsId` and `@assays` slots might actually be useless, because we're adding the respective information as spectra variable. So, yes, I could drop these slots alltogether because we don't need/use them.

Reply:
I removed the slots now. Thanks for pointing me to that.